### PR TITLE
python: Set urgent before early returns

### LINKED ISSUE #28567: [CERT-TEST-FAILURE] IsUrgent flag is not getting propagated on TH scripts
### Feature Area

Area:TimeSync

### Test Case

TC-TIMESYNC-2.10, 2.12, 2.13

### Reproduction steps

Python scripts for TC-TIMESYNC 2.10, 2.12, and 2.13 subscribe to events with a path tuple include 'urgent==1'. However, the DUT receives a subscribe request without the urgent flag set.

This causes said tests to fail on DUT's which increase the maxInterval during the subscription negotiation.

### Bug prevalence

Consistent

### GitHub hash of the SDK that was being used

TH -Fall2023-Beta3

### Platform

python

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by cecille**
Properly set isUrgent on python generated event subscriptions. 

test: subscribed with isUrgent, can now see it being set

Fixes: #28567

### CODE REVIEW

**Review by yunhanw-google — APPROVED** [review]

**Comment by andy31415** [comment]
We seem to get a:

```
2023-08-22T03:24:46.6355506Z [2023-08-22 03:24:46.634621][[32mTEST[0m][[33mSTDOUT[0m][MatterTest] 08-22 03:24:46.627 ERROR Exception occurred in test_TC_ACE_1_2.
2023-08-22T03:24:46.6360367Z [2023-08-22 03:24:46.634673][[32mTEST[0m][[33mSTDOUT[0m]Traceback (most recent call last):
2023-08-22T03:24:46.6361285Z [2023-08-22 03:24:46.634686][[32mTEST[0m][[33mSTDOUT[0m]  File "/__w/connectedhomeip/connectedhomeip/out/venv/lib/python3.9/site-packages/construct/core.py", line 1049, in _build
2023-08-22T03:24:46.6362047Z [2023-08-22 03:24:46.634693][[32mTEST[0m][[33mSTDOUT[0m]    data = struct.pack(self.fmtstr, obj)
2023-08-22T03:24:46.6362682Z [2023-08-22 03:24:46.634698][[32mTEST[0m][[33mSTDOUT[0m]struct.error: required argument is not an integer
2023-08-22T03:24:46.6363406Z [2023-08-22 03:24:46.634703][[32mTEST[0m][[33mSTDOUT[0m]
2023-08-22T03:24:46.6364034Z [2023-08-22 03:24:46.634708][[32mTEST[0m][[33mSTDOUT[0m]During handling of the above exception, another exception occurred:
...
```

**Review by tcarmelveilleux — APPROVED** [review]

File: src/controller/python/chip/ChipDeviceCtrl.py [hunk 1]
```diff
@@ -1019,7 +1019,10 @@ def _parseEventPathTuple(self, pathTuple: typing.Union[
                     event = pathTuple[1]
                 else:
                     raise ValueError("Unsupported Attribute Path")
-                urgent = pathTuple[-1]
+                if len(pathTuple) > 2:
+                    urgent = pathTuple[-1]
```
Comment (tcarmelveilleux) [inline src/controller/python/chip/ChipDeviceCtrl.py:1023]: Suggest:

`urgent = bool(pathTuple[-1]) if len(pathTuple) > 2 else False` or such?

**Review by jrhees-cae — APPROVED** [review]

**Comment by jrhees-cae** [comment]
Note that this is somewhat related to https://github.com/project-chip/connectedhomeip/issues/28817

**Comment by cecille** [comment]
TC-ACE-1.2 isn't setting isUrgent...but it COULD...

**Review by jmartinez-silabs — APPROVED** [review]

File: src/controller/python/chip/ChipDeviceCtrl.py (hunk 1 above)
Comment (cecille) [reply src/controller/python/chip/ChipDeviceCtrl.py]: fixed

**Pull request merged**